lib/ansible/modules/git.py

```diff
@@ -1290,13 +1291,14 @@ def main():
                 if not module.check_mode:
                     relocate_repo(module, result, separate_git_dir, repo_path, dest)
                     repo_path = separate_git_dir
-        except (IOError, ValueError) as err:
+        except (OSError, ValueError) as ex:
             # No repo path found
             # ``.git`` file does not have a valid format for detached Git dir.
             module.fail_json(
                 msg='Current repo does not have a valid reference to a '
                 'separate Git dir or it refers to the invalid path',
-                details=to_text(err),
+                details=to_text(ex),
```

Comment:
No need for `to_text` on an exception, `str` works:

```suggestion
                details=str(ex),
```